pkg/discovery/k8s_discovery_client.go

```diff
@@ -41,30 +41,30 @@ func NewDiscoveryConfig(probeConfig *configs.ProbeConfig,
 type K8sDiscoveryClient struct {
 	config            *DiscoveryClientConfig
 	k8sClusterScraper *cluster.ClusterScraper
-	clusterProcessor  *processor.ClusterProcessor
-	kubeCluster       *repository.KubeCluster
 
-	dispatcher      *worker.Dispatcher
-	resultCollector *worker.ResultCollector
+	clusterProcessor *processor.ClusterProcessor
```

Comment:
Can be concurrency problem because the variables:`clusterProcessor, dispatcher` and `resultCollector` of  `K8sDiscoveryClient`.

The instance of `K8sDiscoveryClient` can be shared for multiple `Discovery` actions. These `Discovery` actions can overlap during their executions.  So these variables are shared, and modified in concurrently, which could cause data corruption.

So it is better to keep `K8sDiscoveryClient` readonly, and put these variables in another agent for each `Discovery` action.

Reply:
We can solve this problem in another PR. The `resultCollector`  has been here since 6.0.

Reply:
K8sDiscoveryClient will handle discovery requests for one instance of target. The worker pool and cluster processor (which holds the cluster API config/client) are persisted across requests. Multiple discovery requests for the same target are queued by the server.